File: add_missing_viewcontrollers.py
#!/usr/bin/env python3
"""Script to add FileExplorerViewController and TerminalViewController to Xcode project"""

import re
import uuid

def generate_xcode_uuid():
    """Generate a 24-character hex UUID for Xcode"""
    return uuid.uuid4().hex[:24].upper()

def add_view_controllers():
    project_file = "/Users/nick/Documents/claude-code-ios-ui/ClaudeCodeUI-iOS/ClaudeCodeUI.xcodeproj/project.pbxproj"
    
    # Read the project file
    with open(project_file, 'r') as f:
        content = f.read()
    
    # Generate UUIDs for FileExplorerViewController
    file_explorer_uuid = generate_xcode_uuid()
    file_explorer_build_uuid = generate_xcode_uuid()
    file_explorer_group_uuid = generate_xcode_uuid()
    
    # Generate UUIDs for TerminalViewController
    terminal_uuid = generate_xcode_uuid()
    terminal_build_uuid = generate_xcode_uuid()
    terminal_group_uuid = generate_xcode_uuid()
    
    # Find where to insert the file references
    # Look for the last swift file reference in the PBXFileReference section
    file_ref_pattern = r'(908A0FC3A7394AEDABD41BAE /\* ChatViewController\.swift \*/ = {isa = PBXFileReference[^}]+};\n)'
    
    # Create the new file references
    new_file_refs = f'''\t\t{file_explorer_uuid} /* FileExplorerViewController.swift */ = {{isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = FileExplorerViewController.swift; sourceTree = "<group>"; }};
\t\t{terminal_uuid} /* TerminalViewController.swift */ = {{isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = TerminalViewController.swift; sourceTree = "<group>"; }};
'''
    
    # Insert the file references
    content = re.sub(file_ref_pattern, r'\1' + new_file_refs, content)
    
    # Find where to add the build files
    # Look for the last build file entry
    build_file_pattern = r'(704436B7DE1D469495491C6D /\* ChatViewController\.swift in Sources \*/ = {isa = PBXBuildFile[^}]+};\n)'
    
    # Create the new build files
    new_build_files = f'''\t\t{file_explorer_build_uuid} /* FileExplorerViewController.swift in Sources */ = {{isa = PBXBuildFile; fileRef = {file_explorer_uuid} /* FileExplorerViewController.swift */; }};
\t\t{terminal_build_uuid} /* TerminalViewController.swift in Sources */ = {{isa = PBXBuildFile; fileRef = {terminal_uuid} /* TerminalViewController.swift */; }};
'''
    
    # Insert the build files
    content = re.sub(build_file_pattern, r'\1' + new_build_files, content)
    
    # Find the Features group and add FileExplorer and Terminal groups
    features_pattern = r'(/\* Features \*/ = {\s+isa = PBXGroup;\s+children = \(\s+CAC97E9AD0264C45900DE34A /\* Chat \*/,)'
    
    # Create group references
    new_groups = f'''\t\t\t\t{file_explorer_group_uuid} /* FileExplorer */,
\t\t\t\t{terminal_group_uuid} /* Terminal */,'''
    
    # Insert group references in Features
    content = re.sub(features_pattern, r'\1\n' + new_groups, content)
    
    # Now add the group definitions
    # Find where to add them (before End PBXGroup section)
    groups_end_pattern = r'(/\* End PBXGroup section \*/)'
    
    groups_def = f'''\t\t{file_explorer_group_uuid} /* FileExplorer */ = {{
\t\t\tisa = PBXGroup;
\t\t\tchildren = (
\t\t\t\t{file_explorer_uuid} /* FileExplorerViewController.swift */,
\t\t\t);
\t\t\tpath = FileExplorer;
\t\t\tsourceTree = "<group>";
\t\t}};
\t\t{terminal_group_uuid} /* Terminal */ = {{
\t\t\tisa = PBXGroup;
\t\t\tchildren = (
\t\t\t\t{terminal_uuid} /* TerminalViewController.swift */,
\t\t\t);
\t\t\tpath = Terminal;
\t\t\tsourceTree = "<group>";
\t\t}};
'''
    
    # Insert the group definitions
    content = re.sub(groups_end_pattern, groups_def + r'\n\1', content)
    
    # Find the Sources build phase and add the files
    # Look for the Sources build phase files section containing our Chat build file
    sources_pattern = r'(704436B7DE1D469495491C6D /\* ChatViewController\.swift in Sources \*/,)'
    
    # Add the build file references to Sources
    new_source_refs = f'''\n\t\t\t\t{file_explorer_build_uuid} /* FileExplorerViewController.swift in Sources */,
\t\t\t\t{terminal_build_uuid} /* TerminalViewController.swift in Sources */,'''
    
    # Insert in the Sources build phase
    content = re.sub(sources_pattern, r'\1' + new_source_refs, content)
    
    # Write the modified content back
    with open(project_file, 'w') as f:
        f.write(content)
    
    print(f"Successfully added FileExplorerViewController and TerminalViewController to Xcode project")
    print(f"FileExplorer UUID: {file_explorer_uuid}")
    print(f"Terminal UUID: {terminal_uuid}")

if __name__ == "__main__":
    add_view_controllers()

File: add_missing_viewcontrollers.rb
#!/usr/bin/env ruby

require 'xcodeproj'

# Open the project
project_path = 'ClaudeCodeUI-iOS/ClaudeCodeUI.xcodeproj'
project = Xcodeproj::Project.open(project_path)

# Find the main target
target = project.targets.find { |t| t.name == 'ClaudeCodeUI' }
raise "Target 'ClaudeCodeUI' not found" unless target

# Find the main app group
main_app_group = project.main_group.children.find do |g| 
  g.is_a?(Xcodeproj::Project::Object::PBXGroup) && g.name != 'Products'
end

puts "Using main app group: #{main_app_group.name || main_app_group.path || 'unnamed'}"

# Find Features group
features_group = main_app_group.children.find do |child|
  child.is_a?(Xcodeproj::Project::Object::PBXGroup) && 
  (child.name == 'Features' || child.path == 'Features')
end

if features_group
  puts "Found Features group"
  
  # Add Search folder and files
  search_group = features_group.children.find do |child|
    child.is_a?(Xcodeproj::Project::Object::PBXGroup) && 
    (child.name == 'Search' || child.path == 'Search')
  end
  
  if search_group
    puts "Found Search group"
    
    # Check and add SearchViewController.swift
    search_vc = search_group.files.find { |f| f.path == 'SearchViewController.swift' }
    unless search_vc
      search_vc = search_group.new_file('SearchViewController.swift')
      target.source_build_phase.add_file_reference(search_vc)
      puts "Added SearchViewController.swift to project"
    else
      puts "SearchViewController.swift already in project"
    end
    
    # Check and add SearchView.swift
    search_view = search_group.files.find { |f| f.path == 'SearchView.swift' }
    unless search_view
      search_view = search_group.new_file('SearchView.swift')
      target.source_build_phase.add_file_reference(search_view)
      puts "Added SearchView.swift to project"
    else
      puts "SearchView.swift already in project"
    end
    
    # Check and add SearchViewModel.swift
    search_vm = search_group.files.find { |f| f.path == 'SearchViewModel.swift' }
    unless search_vm
      search_vm = search_group.new_file('SearchViewModel.swift')
      target.source_build_phase.add_file_reference(search_vm)
      puts "Added SearchViewModel.swift to project"
    else
      puts "SearchViewModel.swift already in project"
    end
    
    # Check and add SearchResultsView.swift
    search_results = search_group.files.find { |f| f.path == 'SearchResultsView.swift' }
    unless search_results
      search_results = search_group.new_file('SearchResultsView.swift')
      target.source_build_phase.add_file_reference(search_results)
      puts "Added SearchResultsView.swift to project"
    else
      puts "SearchResultsView.swift already in project"
    end
  else
    # Create Search group if it doesn't exist
    search_group = features_group.new_group('Search', 'Search')
    puts "Created Search group"
    
    # Add all Search files
    ['SearchViewController.swift', 'SearchView.swift', 'SearchViewModel.swift', 'SearchResultsView.swift'].each do |file|
      file_ref = search_group.new_file(file)
      target.source_build_phase.add_file_reference(file_ref)
      puts "Added #{file} to project"
    end
  end
  
  # Add Git folder and files
  git_group = features_group.children.find do |child|
    child.is_a?(Xcodeproj::Project::Object::PBXGroup) && 
    (child.name == 'Git' || child.path == 'Git')
  end
  
  if git_group
    puts "Found Git group"
    
    # Check and add GitViewController.swift
    git_vc = git_group.files.find { |f| f.path == 'GitViewController.swift' }
    unless git_vc
      git_vc = git_group.new_file('GitViewController.swift')
      target.source_build_phase.add_file_reference(git_vc)
      puts "Added GitViewController.swift to project"
    else
      puts "GitViewController.swift already in project"
    end
    
    # Check and add GitViewModel.swift
    git_vm = git_group.files.find { |f| f.path == 'GitViewModel.swift' }
    unless git_vm
      git_vm = git_group.new_file('GitViewModel.swift')
      target.source_build_phase.add_file_reference(git_vm)
      puts "Added GitViewModel.swift to project"
    else
      puts "GitViewModel.swift already in project"
    end
  else
    # Create Git group if it doesn't exist
    git_group = features_group.new_group('Git', 'Git')
    puts "Created Git group"
    
    # Add all Git files
    ['GitViewController.swift', 'GitViewModel.swift'].each do |file|
      file_ref = git_group.new_file(file)
      target.source_build_phase.add_file_reference(file_ref)
      puts "Added #{file} to project"
    end
  end
  
else
  puts "ERROR: Features group not found!"
end

# Save the project
project.save
puts "Project saved successfully!"